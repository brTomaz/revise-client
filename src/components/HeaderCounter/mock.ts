import { HeaderCounterProps } from '.'

export const mockHeaderCounter: HeaderCounterProps = {
  topText: 'Pontuação',
  mainInfo: 60,
  bottomText: 'pontos'
}
