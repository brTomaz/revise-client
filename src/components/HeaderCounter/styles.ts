import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  text-align: center;
`
export const TopText = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.normal};
  `}
`

export const MainInfo = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xhuge};
    font-weight: ${theme.font.light};
    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxhuge};
    `}
  `}
`
export const BottomText = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.normal};
    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}
  `}
`
