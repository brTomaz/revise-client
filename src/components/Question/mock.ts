/* eslint-disable @typescript-eslint/no-empty-function */
export const mockQuestion = {
  id: 1,
  info:
    '<figure class="image"><img src="/uploads/question_1_bca75dd891.png" alt=""></figure>',
  statement:
    'No cartum, a crítica está no fato de a sociedade exigir do adolescente que ',
  answers: [
    'se aposente prematuramente.',
    'amadureça prematuramente.',
    'estude aplicadamente.',
    'se forme rapidamente.',
    'ouça atentamente.'
  ],
  correctAnswer: 2,
  handleClick: () => {}
}
