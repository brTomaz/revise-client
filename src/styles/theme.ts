export default {
  border: {
    radius: '2.6rem'
  },
  font: {
    family:
      "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '3.2rem',
      xhuge: '4.0rem',
      xxhuge: '4.8rem'
    }
  },
  colors: {
    primary: '#2D3D88',
    secondary: '#2DC2C3',
    darkSecondary: '#166061',
    lightBlue: '#6493EE',
    mainBg: '#06092B',
    lightBg: '#F2F2F2',
    white: '#FAFAFA',
    black: '#030517',
    lightGray: '#EAEAEA',
    gray: '#DADADA',
    darkGray: '#3A3A3A',
    error: '#F74446'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem',
    huge: '6.4rem',
    xhuge: '7.2rem',
    xxhuge: '9.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  }
} as const
