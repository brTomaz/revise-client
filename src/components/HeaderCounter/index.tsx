import * as Styles from './styles'

export type HeaderCounterProps = {
  mainInfo: number
  topText: string
  bottomText: string
}

const HeaderCounter = ({
  mainInfo,
  topText,
  bottomText
}: HeaderCounterProps) => (
  <Styles.Wrapper>
    <Styles.TopText>{topText}</Styles.TopText>
    <Styles.MainInfo>{mainInfo}</Styles.MainInfo>
    <Styles.BottomText>{bottomText}</Styles.BottomText>
  </Styles.Wrapper>
)

export default HeaderCounter
