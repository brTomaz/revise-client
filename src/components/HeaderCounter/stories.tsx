import { Story, Meta } from '@storybook/react/types-6-0'

import HeaderCounter, { HeaderCounterProps } from '.'
import { mockHeaderCounter } from './mock'

export default {
  title: 'HeaderCounter',
  component: HeaderCounter
} as Meta

export const Default: Story<HeaderCounterProps> = (args) => (
  <HeaderCounter {...args} />
)

Default.args = mockHeaderCounter
