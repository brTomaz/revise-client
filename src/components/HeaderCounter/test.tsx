import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'

import HeaderCounter from '.'
import { mockHeaderCounter } from './mock'

describe('<HeaderCounter />', () => {
  it('should render the HeaderCounter with styles', () => {
    const { container } = renderWithTheme(
      <HeaderCounter {...mockHeaderCounter} />
    )

    const topText = screen.getByText(/pontuação/i)
    const mainInfo = screen.getByText(/60/)
    const bottomText = screen.getByText(/pontos/i)

    expect(topText).toHaveStyle({
      'font-size': theme.font.sizes.xlarge,
      'font-weight': 400
    })

    expect(mainInfo).toHaveStyle({
      'font-size': theme.font.sizes.xhuge,
      'font-weight': 100
    })

    expect(mainInfo).toHaveStyleRule('font-size', theme.font.sizes.xxhuge, {
      media: '(min-width: 768px)'
    })

    expect(bottomText).toHaveStyle({
      'font-size': theme.font.sizes.medium,
      'font-weight': 300
    })

    expect(bottomText).toHaveStyleRule('font-size', theme.font.sizes.large, {
      media: '(min-width: 768px)'
    })

    expect(container).toMatchSnapshot()
  })
})
