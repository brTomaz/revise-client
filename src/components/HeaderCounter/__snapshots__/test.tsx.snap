// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderCounter /> should render the HeaderCounter with styles 1`] = `
.c0 {
  text-align: center;
}

.c1 {
  font-size: 2.0rem;
  font-weight: 400;
}

.c2 {
  font-size: 4.0rem;
  font-weight: 100;
}

.c3 {
  font-size: 1.6rem;
  font-weight: 400;
}

@media (min-width:768px) {
  .c2 {
    font-size: 4.8rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 1.8rem;
  }
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      Pontuação
    </h2>
    <div
      class="c2"
    >
      60
    </div>
    <div
      class="c3"
    >
      pontos
    </div>
  </div>
</div>
`
